pkg/reconciler/servingpipeline/controller.go

```diff
@@ -189,6 +191,13 @@ func (r *servingPipelineReconciler) reconcileFixedResources(ctx context.Context,
 		r.recorder.Eventf(spl, corev1.EventTypeWarning, "CreateOrUpdateServingServerFailed", "Failed to create or update serving server: %w", err.Error())
 		return fmt.Errorf("failed to create or update serving server: %w", err)
 	}
+
+	if err := r.createOrUpdateServingHPA(ctx, spl); err != nil {
```

Comment:
Please do NOT create HPA on behalf.

Reply:
Let the user manage the autoscaling strategy by themselves - could use HPA, or something else. Here, we might introduce a field `replicas` to control the initial replicas.

Reply:
Should we use the `spec.serving.scale.min` to set the replicas (`max` will be ignored) or should we introduce a `spec.serving.replicas` field ?
```yaml
spec:
  serving:
    service: true
    msgIDHeaderKey: "X-Numaflow-Id"
    containerTemplate:
      resources:
        requests:
          cpu: 1
          memory: 1Gi
    scale:
      min: 2
      max: 10
```

Reply:
Going to add `spec.serving.replicas` field. It will only have an effect when deployment is created for the first time. Since there could be a user managed HPA, I don't think we should update the deployment objects replicas other than during initial creation.

Reply:
Right. Exclude the `replicas` when generating the hash.

Reply:
Does this work:
```go
	// Number of replicas. If an HPA is used to manage the deployment object, do not set this field.
	// +optional
	Replicas *int32 `json:"replicas,omitempty" protobuf:"varint,7,opt,name=replicas"`
```
And then simply use the `deploymentObject.Replicas = spl.Spec.Serving.Replicas`

Reply:
This way, if the user doesn't have an HPA, and wanted to scale down to a lower value, it works

Reply:
hmmm.. take it back - `Exclude the replicas when generating the hash.` I think we still need to include replicas in the hash.